ktor-server/ktor-server-plugins/ktor-server-openapi/jvm/src/io/ktor/server/openapi/OpenAPI.kt

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2014-2022 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.server.openapi
+
+import io.ktor.server.application.*
+import io.ktor.server.html.*
+import io.ktor.server.http.content.*
+import io.ktor.server.request.*
+import io.ktor.server.response.*
+import io.ktor.server.routing.*
+import io.swagger.codegen.v3.*
+import io.swagger.codegen.v3.generators.html.*
+import io.swagger.parser.*
+import io.swagger.v3.parser.core.models.*
+import kotlinx.html.*
+import java.io.*
+
+/**
+ * Create a get endpoint with [SwaggerUI] ath [path] rendered from open api file at [filename].
+ */
+public fun Routing.swaggerUI(path: String, filename: String) {
+    route(path) {
+        get("api.json") {
+            val file = File(filename)
+            call.respondFile(file)
+        }
+        get {
+            val fullPath = call.request.path()
+            call.respondHtml {
+                head {
+                    title { +"Swagger UI" }
+                    link(rel = "stylesheet", href = "https://unpkg.com/swagger-ui-dist@4.5.0/swagger-ui.css")
+                }
+                body {
+                    div { id = "swagger-ui" }
+                    script(src = "https://unpkg.com/swagger-ui-dist@4.5.0/swagger-ui-bundle.js") {
+                        attributes["crossorigin"] = "anonymous"
+                    }
+                    script(src = "https://unpkg.com/swagger-ui-dist@4.5.0/swagger-ui-standalone-preset.js") {
+                        attributes["crossorigin"] = "anonymous"
+                    }
+
+                    script {
+                        unsafe {
+                            +"""
+window.onload = function() {
+    window.ui = SwaggerUIBundle({
```

Comment:
Also, as I remember swagger UI has options to configure it look and feel, that could be exposed for customization

Reply:
Nice idea, will check